src/features/search/helpers/useSearchHistory/useSearchHistory.tsx

```diff
@@ -0,0 +1,111 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+import { getHistoryLessThan30Days } from 'features/search/helpers/useSearchHistory/helpers/getHistoryLessThan30Days'
+import { HistoryItem } from 'features/search/types'
+import { eventMonitoring } from 'libs/monitoring'
+import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
+
+export function useSearchHistory() {
+  const { showErrorSnackBar } = useSnackBarContext()
+  const [history, setHistory] = useState<HistoryItem[]>([])
+  const [queryHistory, setQueryHistory] = useState<string>('')
+
+  const setHistoryItems = useCallback(async (newItems: HistoryItem[]) => {
+    return AsyncStorage.setItem('search_history', JSON.stringify(newItems))
+  }, [])
+
+  const getHistoryFromStorage = useMemo(
+    () => async () => {
+      try {
+        const items = (await AsyncStorage.getItem('search_history')) ?? '[]'
+        const history: HistoryItem[] = JSON.parse(items)
+
+        const historyLessThan30Days = getHistoryLessThan30Days(history)
+        await setHistoryItems(historyLessThan30Days)
+
+        return historyLessThan30Days
+      } catch (error) {
+        return []
+      }
+    },
+    [setHistoryItems]
+  )
```

Comment:
Bizarre de même ça dans un `useMemo` ?

```suggestion
  const getHistoryFromStorage = useCallback(async () => {
    try {
      const items = (await AsyncStorage.getItem('search_history')) ?? '[]'
      const history: HistoryItem[] = JSON.parse(items)

      const historyLessThan30Days = getHistoryLessThan30Days(history)
      await setHistoryItems(historyLessThan30Days)

      return historyLessThan30Days
    } catch (error) {
      return []
    },
    [setHistoryItems]
  )
```